saleor/graphql/utils.py

```diff
@@ -20,6 +24,49 @@ def get_database_id(info, node_id, only_type):
     return _id
 
 
+def _check_graphene_type(requested_graphene_type, received_type):
+    if requested_graphene_type:
+        assert str(requested_graphene_type) == received_type, (
+            'Must receive an {} id.'
+        ).format(requested_graphene_type._meta.name)
+
+
+def _resolve_nodes(ids, graphene_type=None):
+    pks = []
+    types = []
+    invalid_ids = []
+
+    for graphql_id in ids:
+        if not graphql_id:
+            continue
+
+        try:
+            _type, _id = from_global_id(graphql_id)
+        except Exception:
+            invalid_ids.append(graphql_id)
+            continue
+
+        _check_graphene_type(graphene_type, _type)
+        pks.append(_id)
+        types.append(_type)
+
+    if invalid_ids:
+        raise GraphQLError(
+            ERROR_COULD_NO_RESOLVE_GLOBAL_ID % invalid_ids)
+
+    return types, pks
+
+
+def _resolve_graphene_type(types: List):
+    assert len(set(types)) == 1, 'Received IDs of more than one type.'
+    # get type by name
+    type_name = types[0]
+    for _, _type in registry._registry.items():
+        if _type._meta.name == type_name:
+            return _type
+    return None
```

Comment:
I can't recall a case when we may want to get `None` here - so maybe raising an ValidationError should be present here.

Reply:
I decided to raise a `AssertionError`, is that okay? It seems to fit better in the context.